api: GetPullRequestCommits: return file list

msantos opened the pull request:
Fixes https://github.com/go-gitea/gitea/issues/27481

---
Patch tested:

```json
[
  {
    "url": "http://100.115.92.198:9292/api/v1/repos/msantos/test/git/commits/7664dcb44167e0f9efd994e4ca6a9164694adc27",
    "sha": "7664dcb44167e0f9efd994e4ca6a9164694adc27",
    "created": "2023-10-06T09:57:08-04:00",
    "html_url": "http://100.115.92.198:9292/msantos/test/commit/7664dcb44167e0f9efd994e4ca6a9164694adc27",
...
    "files": [
      {
        "filename": "README.md",
        "status": "modified"
      }
    ],
    "stats": {
      "total": 2,
      "additions": 2,
      "deletions": 0
    }
  }
]
```

silverwind commented:
Was this a regression? `files: null` does indicate i may have been.

msantos commented:
> Was this a regression? `files: null` does indicate i may have been. If there are tests for this api call, I suggest validating presence of `files`.

Yes, I run some status checks using `files`. I'd have to check but I think it was working in 1.19. I'll look into the tests.

silverwind commented:
Thanks, maybe you can also check the 1.20 and 1.21 branches so we know how far to backport this.

msantos commented:
> Thanks, maybe you can also check the 1.20 and 1.21 branches so we know how far to backport this.

Introduced in https://github.com/go-gitea/gitea/commit/1dd83dbb917d55bd253001646d6743f247a4d98b released in 1.20.

I see the above commit made returning files/verification optional for the commit endpoint by using a parameter:

```
curl -X 'GET'   'https://try.gitea.io/api/v1/repos/msantos/test/commits?files=true'   -H 'accept: application/json'
````

The commit changed 3 endpoints: commit, pull and notes.

* pulls don't include verification
* notes don't include files or verification

Is it ok to have pulls always return files or should it be a paramter?

msantos commented:
Updated to:
* return files/verification fields by default in the pulls and notes endpoints
* optionally disable using the `fields` and `verification` paramaters
* added tests

wxiaoguang commented on routers/api/v1/repo/notes.go:
```diff
@@ -78,7 +78,15 @@ func getNote(ctx *context.APIContext, identifier string) {
 	}
 
-	cmt, err := convert.ToCommit(ctx, ctx.Repo.Repository, ctx.Repo.GitRepo, note.Commit, nil, convert.ToCommitOptions{Stat: true})
+	verification := ctx.FormString("verification") == "" || ctx.FormBool("verification")
+	files := ctx.FormString("files") == "" || ctx.FormBool("files")
```
Here and below, have these two new parameters been documented in swagger comment?

msantos replied:
Thanks! Fixed.

silverwind commented:
Given that previous `files` api was undocumented, I think we can strictly speaking not call it a regression.

msantos commented:
> Given that previous `files` api was undocumented, I think we can strictly speaking not call it a regression, but I'd be willing to have it in 1.21.

https://github.com/go-gitea/gitea/commit/1dd83dbb917d55bd253001646d6743f247a4d98b added the new `files` api to the commits endpoint in 1.20 by modifying a function shared by the notes and pulls endpoints. The change unconditionally disabled returning values for the `files` and `verification` fields.

The change in this PR:
* notes/pulls: restores the behaviour in 1.19 of returning files/verification
* notes/pulls: allows optionally disabling files/verification by copying the `files` api from https://github.com/go-gitea/gitea/commit/1dd83dbb917d55bd253001646d6743f247a4d98b

The second item is unrelated to https://github.com/go-gitea/gitea/issues/27481 and could be dropped or split into a separate PR. Sorry for any confusion!
